### PR TITLE
Support sub-sub directory multisite file uploads

### PR DESCRIPTION
**Pull request by darthhexx**
Add's regex checks and path substitutions for sub-sub directory multisite file uniqueness checks and uploads.

### CODE REVIEW

File: a8c-files.php [hunk 1]
```diff
@@ -276,7 +280,7 @@ public function filter_unique_filename( $filename, $ext, $dir, $unique_filename_
 
 		if ( 200 == $check['http_code'] ) {
 			$obj = json_decode( $check['content'] );
-			if ( isset(  $obj->filename ) && basename( $obj->filename ) != basename( $post_url ) ) {
+			if ( isset(  $obj->filename ) && basename( $obj->filename ) != basename( $filename ) ) {
```
Comment (mjangda) [inline a8c-files.php]: Good catch!

**Comment by mjangda** [comment]
Thanks @darthhexx; the updates looked good.

I abstracted the code into a new class to clean things up a bit (the code in `a8c-files` is getting a bit out-of-control :)) and so we could add some unit tests. Let me know if my changes look okay.

**Comment by darthhexx** [comment]
LGTM :shipit:

**Comment by mjangda** [comment]
r113559-deploy

**Pull request merged**